.maestro/video-downloader/cypress/e2e/downloadVideos.cy.js

```diff
@@ -0,0 +1,101 @@
+describe('Login, collect tests & download videos', () => {
+
+  const email = Cypress.env('maestroEmail');
+  const projectUrl = Cypress.env('projectUrl');
+
+  it('Login, collect test names, download videos', {
+    defaultCommandTimeout: 90000,
+    execTimeout: 180000,
+    taskTimeout: 600000
+  }, () => {
+
+    //
+    // --- Step 1: Login ---
+    //
+    cy.visit('https://signin.maestro.dev');
+    cy.get('input[name="email"]').type(email);
+    cy.get('button[type="submit"]').click();
+
+    // Get *newest* OTP — retry until latest arrives
+    cy.task('getMaestroOtp', { retry: true }).then((otp) => {
+      expect(otp, 'OTP should not be null').to.not.be.null;
+      cy.get('input[data-test="otp-input"]').first().type(otp, { delay: 100 });
+    });
+
+    //
+    // --- Step 2: Enter Maestro Dashboard ---
+    //
+    cy.origin('https://app.maestro.dev', { args: { projectUrl } }, ({ projectUrl }) => {
+
+      cy.wait(6000);
```

Comment:
NO: a) We must only use a "wait" command when there is absolutely no viable alternative
b) When there is absolutely no viable alternative there MUST be a source code comment explaining the situation in the (rare) instances it cannot be avoided.